hooks/useMqlQuery/useMqlQuery.ts

```diff
@@ -44,27 +45,27 @@ export default function useMqlQuery({
   const reducer = mqlQueryReducer<CreateMqlQueryMutation>((data: CreateMqlQueryMutation) => data?.createMqlQuery?.query);
   const [state, dispatch] = useReducer(reducer, initialState);
   const {createMqlQuery} = useCreateMqlQuery({metricName, formState: queryInput, dispatch, retries})
+  const [queryReruns, setQueryReruns] = useState(0);
 
   useEffect(() => {
     if (!metricName || !mqlServerUrl || skip) {
       return;
+    } else if (!state.queryId) {
+      dispatch({ type: "postQueryStart" }); 
+      createMqlQuery({stateRetries: state.retries});
+    } else if (state.queryId && !state.cancelledQueries.includes(state.queryId)) {
+      useFetchMqlTimeSeries<CreateMqlQueryMutation>({
+        state,
+        skip: false,
+        dispatch,
+        createQueryIdQuery: createMqlQuery,
+        retries
+      });
     }
-    dispatch({ type: "postQueryStart" });
-    createMqlQuery({stateRetries: state.retries});
-  }, [queryInput, metricName, mqlServerUrl, skip]);
-
-  const _skip =
-    skip ||
-    !state.queryId ||
-    state.cancelledQueries.includes(state.queryId); /* && !isRunning*/
-
-  useFetchMqlTimeSeries<CreateMqlQueryMutation>({
-    state,
-    skip: _skip,
-    dispatch,
-    createQueryIdQuery: createMqlQuery,
-    retries
-  });
+  }, [queryInput, metricName, mqlServerUrl, skip, queryReruns]);
 
+  if (state.queryStatus === MqlQueryStatus.Pending || state.queryStatus === MqlQueryStatus.Running) {
```

Comment:
Random thought... is it possible there's an in between state here that we're not picking up in these checks?

I.e. maybe another query status or the query status is undefined?

Reply:
I don't think so, query status is directly pulled from the query result returned from the MQL server and we can see in the network responses that it's usually RUNNING when the retries stop.

Reply:
Yeah, there's no other state that we need to account for here. My theory is that there's no update to the parameters to the `useEffect`, so it fails to check again.